### DISCUSSION TITLE
Interesting behaviour/interaction between matplotlib and validation_epoch_end(self, outputs)

### DISCUSSION THREAD

**Discussion by nick-torenvliet** [Lightning-AI/pytorch-lightning#13282] [code help: CV] [answered]
I am modifying some code taken out of the lightning docs and training a VAE to learn a represenation of time series thought to contain structure.  The validation outputs are reconstructions of the time series.  The following code is written with the intention of grabbing a random output and plotting it.

Two things happen that I don't understand. 

1 - It seems the plt object from the previous epoch is still in memory and the new data is plotted alongside the old.  I have tried deleting the data explicitly but to no avail... 
2 - Somehow both the input and the output are getting plotted (in the attached plot you see the learned noise, and the well formed actual singal I am trying to reconstruct. I have no idea where the input signal is coming from - is it embedded in the output tensor somewhere?
 
    def validation_epoch_end(self, outputs):
        if not self.save_images:
            return
        if not os.path.exists(self.save_path):
            os.makedirs(self.save_path)
        choice = random.choice(outputs)
        output_sample = random.choice(choice[0])
        plt.plot(output_sample.cpu())
        plt.savefig(f"{self.save_path}/epoch_{self.current_epoch}.png")
        print('this is def interesting')

![epoch_0](https://user-images.githubusercontent.com/25708723/173474539-a50d4833-debe-4a69-a777-5aff7b555132.png)
![epoch_1](https://user-images.githubusercontent.com/25708723/173474546-81e00883-e6e1-43eb-b2ac-de025024df34.png)
![epoch_2](https://user-images.githubusercontent.com/25708723/173474554-d2e35885-4080-423c-97d5-1fb51e78e320.png)

**Comment by semaphore-egg** [comment, marked as answer]
Creating a new figure before plotting?
```python
fig, ax = plt.subplots()
ax.set_xlim(0, 10)
ax.plot([1, 5], [0, 10])
```

**Reply by nick-torenvliet** [reply]
Yeah - that will work... I'm just suprised the plot object, which is only referenced from inside a
 function call,  is persistent across function calls.
